src/main/java/uk/gov/hmcts/dts/fact/controllers/admin/AdminCourtTypesAndCodesController.java

```diff
@@ -0,0 +1,71 @@
+package uk.gov.hmcts.dts.fact.controllers.admin;
+
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import uk.gov.hmcts.dts.fact.config.security.Role;
+import uk.gov.hmcts.dts.fact.model.admin.CourtType;
+import uk.gov.hmcts.dts.fact.model.admin.CourtTypesAndCodes;
+import uk.gov.hmcts.dts.fact.services.admin.AdminCourtTypesAndCodesService;
+
+import java.util.List;
+
+import static org.springframework.http.ResponseEntity.ok;
+import static uk.gov.hmcts.dts.fact.services.admin.AdminRole.FACT_ADMIN;
+import static uk.gov.hmcts.dts.fact.services.admin.AdminRole.FACT_SUPER_ADMIN;
+
+@RestController
+@RequestMapping(
+    path = "/admin/courts",
+    produces = {MediaType.APPLICATION_JSON_VALUE}
+)
+public class AdminCourtTypesAndCodesController {
+    private final AdminCourtTypesAndCodesService courtTypesAndCodesService;
+
+    @Autowired
+    public AdminCourtTypesAndCodesController(AdminCourtTypesAndCodesService adminService) {
+        this.courtTypesAndCodesService = adminService;
+    }
+
+    @GetMapping(path = "/courtTypes")
+    @ApiOperation("Return all court types")
+    @ApiResponses(value = {
+        @ApiResponse(code = 200, message = "Successful", response = CourtType.class),
+        @ApiResponse(code = 401, message = "Unauthorized"),
+        @ApiResponse(code = 403, message = "Forbidden"),
+    })
+    @Role({FACT_ADMIN, FACT_SUPER_ADMIN})
+    public ResponseEntity<List<CourtType>> getAllCourtTypes() {
+        return ok(courtTypesAndCodesService.getAllCourtTypes());
+    }
+
+    @GetMapping(path = "/{slug}/courtTypesAndCodes")
+    @ApiOperation("Find a court's types and codes by slug")
```

Comment:
Optional: for the swagger, may be clearer to mention that codes refer to the DX (one or many) and GBS (one) codes

Reply:
updated swagger description